conjur/init/init_controller.py

```diff
@@ -45,24 +45,56 @@ def load(self):
         self.write_conjurrc()
 
         sys.stdout.write("Configuration initialized successfully!\n")
-        sys.stdout.write("To begin using the CLI, log in to the Conjur server by" \
-                         "running `conjur authn login`")
+        sys.stdout.write("To begin using the CLI, log in to the Conjur server by " \
+                         "running `conjur login`\n")
 
     def get_server_certificate(self):
         """
         Method to get the certificate from the Conjur endpoint detailed by the user
         """
         # pylint: disable=line-too-long
         if self.conjurrc_data.appliance_url is None:
-            self.conjurrc_data.appliance_url = input("Enter the URL of your Conjur server: ")
+            self.conjurrc_data.appliance_url = input("Enter the URL of your Conjur server: ").strip()
+            if self.conjurrc_data.appliance_url is None or self.conjurrc_data.appliance_url == '':
+                # pylint: disable=raise-missing-from
+                raise RuntimeError("Error: URL is required\n")
+
 
         url = urlparse(self.conjurrc_data.appliance_url)
+        # TODO: Factor out the following URL validation to ConjurrcData class
+        # pylint: disable=pointless-string-statement
+
+        # This condition catches if a user provides input <hostname>
+        # without HTTP/HTTPS prefix. If so, we default the behavior
+        # to HTTPS because this is what we currently support
+        if url.scheme == '':
+            parsed_url = urllib.parse.urlparse(self.conjurrc_data.appliance_url)
+            replaced_url = parsed_url._replace(scheme="https", netloc=url.path, path='')
+            url = replaced_url
+            self.conjurrc_data.appliance_url=replaced_url.geturl()
+
         if url.scheme == "http":
             return None
 
+        # pylint: disable=pointless-string-statement
+
+        # This condition catches the undefined behavior when a user inputs
+        # <hostname>:443 without HTTP/HTTPS prefix. If we don't catch this here,
+        # the user will fail when they try to fetch Conjur resources
+        if (url.scheme != '' and url.path != ''):
+            raise RuntimeError(f"Error: undefined behavior. Reason: The Conjur url format provided "
+                               f"'{self.conjurrc_data.appliance_url}' is not supported. "
+                               "Consider adding HTTPS as the prefix\n")
+
         if self.conjurrc_data.cert_file is not None:
             # Return None because we do not need to fetch the certificate
             return None
+
+        # To follow suit with the Ruby CLI, we don't write the url and the port
+        # in the conjurrc. If the port is added to the conjurrc following requests
+        # will fail
+        self.conjurrc_data.appliance_url = url.scheme + '://' + url.hostname
```

Comment:
You can't ignore the port, removing the port means you refer to different location than the one specified.
If you only support portless http/https, validate it and fail, so you won't need to write this kind of code.

Reply:
The Ruby CLI doesn't add the port to the conjurrc. I use that port to fetch the certificate 
Should we just not support a port?

Reply:
yes. Validate no port is supplied, fail it if it does, and do not build the URL yourself.

Reply:
What are the consequences of such a decision?